modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1876,6 +1878,105 @@ public void testGetObjectEntriesAggregationFacets() throws Exception {
 		Assert.assertEquals(facetValue.getNumberOfOccurrences(), (Integer)2);
 	}
 
+	@Test
+	public void testGetRelatedObjectEntriesAccountEntryRestrictions()
+		throws Exception {
+
+		ObjectDefinition objectDefinition = _createObjectDefinition(
+			Arrays.asList(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"textObjectFieldName"
+				).build()));
+
+		String objectRelationship1Name = "oneToManyRelationshipParentChild";
+
+		ObjectRelationship objectRelationship1 =
+			_objectRelationshipLocalService.addObjectRelationship(
+				_adminUser.getUserId(),
+				_objectDefinition3.getObjectDefinitionId(),
+				objectDefinition.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				objectRelationship1Name,
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		ObjectDefinition accountEntryObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				_companyId, "AccountEntry");
+
+		ObjectRelationship objectRelationship2 =
+			_objectRelationshipLocalService.addObjectRelationship(
+				_adminUser.getUserId(),
+				accountEntryObjectDefinition.getObjectDefinitionId(),
+				objectDefinition.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"oneToManyRelationshipAccountChild",
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		objectDefinition.setAccountEntryRestrictedObjectFieldId(
+			objectRelationship2.getObjectFieldId2());
+
+		objectDefinition.setAccountEntryRestricted(true);
+
+		objectDefinition = _objectDefinitionLocalService.updateObjectDefinition(
+			objectDefinition);
+
+		_addRelatedObjectEntries(
+			_objectDefinition3, objectDefinition, "externalReferenceCode1",
+			"externalReferenceCode2", objectRelationship1);
+
+		AccountEntry accountEntry = _addAccountEntry();
+
+		AccountRole accountRole = _accountRoleLocalService.addAccountRole(
+			TestPropsValues.getUserId(), accountEntry.getAccountEntryId(),
+			RandomTestUtil.randomString(), null, null);
+
+		User user = _addUser();
+
+		_accountEntryUserRelLocalService.addAccountEntryUserRel(
+			accountEntry.getAccountEntryId(), user.getUserId());
+
+		_accountRoleLocalService.associateUser(
+			accountEntry.getAccountEntryId(), accountRole.getAccountRoleId(),
+			user.getUserId());
+
+		Role role = accountRole.getRole();
+
+		_addResourcePermission(ActionKeys.VIEW, role);
+
+		_resourcePermissionLocalService.addResourcePermission(
+			_companyId, objectDefinition.getClassName(),
+			ResourceConstants.SCOPE_GROUP_TEMPLATE, "0", role.getRoleId(),
+			ActionKeys.VIEW);
```

Comment:
@pedro-oliveira446 can we use the `_addResourcePermission` method?

Reply:
@MarinhoFeliphe, only if we change the method, and the places that use it, because in this case I need to add the permissions for the 'childObjectDefinition' and not for the '_objectDefinition3'.